docs/usage/automerge-configuration.md

```diff
@@ -0,0 +1,148 @@
+---
+title: Automerge configuration and troubleshooting
+description: Learn all about Renovate's automerge functionality here
+---
+
+# Introduction
+
+Automerging is a Renovate feature that can enable you to fully automate upgrading of certain dependencies.
+When enabled, it means Renovate will attempt to merge the proposed update once it has passed tests.
+
+As a general guide, it is recommended that you enable automerge for any types of dependency updates where you would just click Merge anyway.
+If there are updates for which you prefer to manually review release notes or code before merging, even when they pass tests, then do not enable automerge for those.
+
+Automerge works particularly well for `devDependencies` as well as for production `dependencies` in projects which have great test coverage.
+
+For example, if you have `jest` or `mocha` as a dependency, and it has an upgrade with passing tests... automerge them!
+If you have a linter like `eslint` or `tslint` and its update passes... automerge them!
+If you have an API with 100% test coverage and `express` is updated... automerge it!
+
+## Configuration examples
+
+<!-- Internal comment
+
+I think the endgame here should be:
+
+- Each example use case is fully realised, and should be copy/pasteable and work.
+- The example should explain the reasoning/thinking behind each setting.
+- The example should have links to each setting, so that the docs can be consulted.
+
+-->
+
+### Automerge lock file maintenance
+
+The lowest risk type of update to automerge is probably `lockFileMaintenance`.
+When Renovate performs lock file maintenance, it means that it leaves the project dependency definitions unchanged, but refreshes the lock file completely so that latest versions according to package file constraints are installed.
+Here is an example of automerging lock file maintenance:
+
+```json
+{
+  "lockFileMaintenance": {
+    "enabled": true,
+    "automerge": true
+  }
+}
+```
+
+### Automerge lint tool updates
+
+Automerging lint tool updates can be a real time-saver.
+Sometimes an update to a lint tool or plugin definition can cause tests to fail, and that is usually deliberate/intentional because the lint authors have added a new validation that you need to adhere to.
+However, in many cases the new version(s) will pass tests, and if so then there's really nothing else to consider before merging, so they may as well be automerged:
+
+```json
+{
+  "packageRules": [
+    {
+      "depTypeList": ["devDependencies"],
+      "packagePatterns": ["lint", "prettier"],
+      "automerge": true
+    }
+  ]
+}
+```
+
+### Automerge non-major updates
+
+Non-major updates in SemVer ecosystems shouldn't have breaking changes (if they follow the spec), therefore many users enable automerge for these too:
+
+```json
+{
+  "packageRules": [
+    {
+      "updateTypes": ["minor", "patch"],
+      "matchCurrentVersion": "!/^0/",
+      "automerge": true
+    }
+  ]
+}
+```
```

Comment:
@rarkins https://docs.renovatebot.com/configuration-options/#matchcurrentversion says you can use this:

```json
{
  "packageRules": [
    {
      "packagePatterns": ["io.github.resilience4j"],
      "allowedVersions": "!/^0\\./"
    }
  ]
}
```

So I was wondering if the new example in the automerge guide is the right/best way? Or is there more than one valid way to do the exclusion of pre-1.0.0 versions?

What's the real difference between `"matchCurrentVersion": "!/^0/"` and `"allowedVersions": "!/^0\\./"`?
I'm confused because it seems matchCurrentVersion and allowedVersions are doing kinda the same thing?

The docs say that `matchCurrentVersion` deals with semver versions or semver range, and that `allowedVersions` is for capturing allowed versions for dependencies?

Also the block talking about `matchCurrentVersion` suddently use `allowedVersions` in a code example...
Should I open a new issue to discuss this further?

Reply:
These are two different things. The first one is for automerging versions only when the existing version is >=1.0.0. The one you mention is a little weird and says don't allow NEW versions starting with `0`.

`matchCurrentVersion` is a matching rule, used to decide whether this package rule applies. `allowedVersions` is an instruction on how to filter the list of possible versions to update to.

Reply:
> `matchCurrentVersion` is a matching rule, used to decide whether this package rule applies. `allowedVersions` is an instruction on how to filter the list of possible versions to update to.

Ok. :+1: I'll resolve this conversation.

@rarkins, can you please also take a look at the review comments I've left in this [PR's file tab](https://github.com/renovatebot/renovate/pull/7724/files) and commit what you like best?